### ISSUE TITLE
[External Application Button] Add option to pass `shell` spawn option

### ISSUE THREAD

**Issue by welwood08** [andy-portmen/native-client#17]
I've been struggling to get youtube-dl, a command-line application, to show me what it's doing when launched through this addon. On my system a new console opens for each of its subprocesses, the previous console immediately closes, and no output appears even though it is downloading.

Adding `shell: true` to the options passed to `child_process.spawn` in host.js is all I needed to get it to use 1 console window and actually display output. That solution has side-effects though, so is it possible this could instead be added as an option for each application on the addon side please?

**Comment by andy-portmen** [comment]
What is your OS and what command are you using?

**Comment by welwood08** [comment]
I'm on Windows 8.1, executable name was set to `C:\ProgramData\chocolatey\lib\youtube-dl\tools\youtube-dl.exe` with arguments `[HREF]`.

Approaching this again now, it seems the problem goes away if I use an executable name of either `C:\ProgramData\chocolatey\bin\youtube-dl.exe` or simply `youtube-dl.exe` - I'm not sure why I had selected the other path originally. For background, Chocolatey is a package manager for Windows, it adds that bin directory to the path, packages are extracted into the lib directory and for any new exe there it generates a "shim" exe in bin that somehow executes the real exe in lib. youtube-dl launches among other things ffmpeg, which I've also installed through Chocolatey and therefore has a similar "shim" exe on path. So I guess it's some weird interaction between the way node directly launched the real youtube-dl and how that then launched the Chocolatey shims (which in turn launched an appropriate real exe).

TL;DR: I don't think I need the shell option anymore, but feel free to use the idea if you think it'd be useful anyway. Sorry for the noise and thanks for such a generic solution for launching things from a webext!
